src/python/turicreate/toolkits/object_detector/_tf_image_augmenter.py

```diff
@@ -41,39 +33,36 @@
   'min_eject_coverage': 0.5
 }
 
-def hue_augmenter(image, annotation,
+def hue_augmenter(image, annotation, tf_seed,
                   max_hue_adjust=_DEFAULT_AUG_PARAMS["max_hue_adjust"]):
 
-    # Sample a random rotation around the color wheel.
-    hue_adjust = 0.0
-    if (max_hue_adjust is not None) and (max_hue_adjust > 0.0):
-        hue_adjust += np.pi * np.random.uniform(-max_hue_adjust, max_hue_adjust)
-
     # Apply the rotation to the hue
-    image = tf.image.random_hue(image, max_delta=max_hue_adjust)
+    image = tf.image.random_hue(image, max_delta=max_hue_adjust, seed=tf_seed)
     image = tf.clip_by_value(image, 0, 1)
 
     # No geometry changes, so just copy the annotations.
     return image, annotation
 
-def color_augmenter(image, annotation,
+
+
+def color_augmenter(image, annotation, tf_seed,
                     max_brightness=_DEFAULT_AUG_PARAMS["max_brightness"],
                     max_contrast=_DEFAULT_AUG_PARAMS["max_contrast"],
                     max_saturation=_DEFAULT_AUG_PARAMS["max_saturation"]):
 
     # Sample a random adjustment to brightness.
     if max_brightness is not None and max_brightness > 0:
-        image = tf.image.random_brightness(image, max_delta=max_brightness)
+        image = tf.image.random_brightness(image, max_delta=max_brightness, seed=tf_seed)
 
     # Sample a random adjustment to contrast.
     if max_saturation is not None and max_saturation > 1.0:
         log_sat = np.log(max_saturation)
-        image = tf.image.random_saturation(image, lower=np.exp(-log_sat), upper=np.exp(log_sat))
+        image = tf.image.random_saturation(image, lower=np.exp(-log_sat), upper=np.exp(log_sat), seed=tf_seed)
```

Comment:
Here you are using same seed as for the `random_brightness`. In the TF implementation (https://github.com/tensorflow/tensorflow/blob/cf7fcf164c9846502b21cebb7d3d5ccf6cb626e8/tensorflow/python/ops/image_ops_impl.py#L1582), I see that this is used as
```
delta = random_ops.random_uniform([], -max_delta, max_delta, seed=seed)
```
Will the delta used for brightness be correlated with the offset used for e.g. saturation? The end of the `set_seed` TF docs suggest that in some cases, ops may or may not share the same internal state.

Reply:
Oh cool. I'll take a look.

Reply:
Haha actually will.
As long as two random tensors are declared with the same type and the same seed,
if they are ran by the same session,
they produce the same result haha.